src/framework/vst/internal/vstaudioclient.cpp

```diff
@@ -140,6 +140,15 @@ audio::samples_t VstAudioClient::process(float* output, audio::samples_t samples
 
     m_processData.numSamples = samplesPerChannel;
 
+    //! NOTE: From the VST3 documentation:
+    //!
+    //! Note that the ProcessData->numSamples
+    //! which indicates how many samples are used in a process call can change from call to call,
+    //! but never bigger than the maxSamplesPerBlock
+    if (samplesPerChannel != m_samplesInfo.maxSamplesPerBlock) {
```

Comment:
why not `samplesPerChannel < m_samplesInfo.maxSamplesPerBlock`?

Reply:
maxSamplesPerBlock may be bigger or smaller depending on the current situation:

* when entering notes it is 1024
* when exporting an mp3 file it is 2048

This check optimizes such cases and not only increases maxSamplesPerBlock, but also decreases it (to store smaller buffers) depending on the situation

Reply:
The name is confusing. `max` - usually means the maximum limit. Perhaps it means that `maxSamplesPerBlock` should be named like `samplesPerBlock`